src/Form/DataTransformer/ModelsToArrayTransformer.php

```diff
@@ -242,4 +239,36 @@ private function triggerDeprecation(): void
             __CLASS__
         ), E_USER_DEPRECATED);
     }
+
+    /**
+     * @param array|\Traversable $value
+     *
+     * @throws \TypeError
+     *
+     * @return Collection<int|string, object>
+     *
+     * @phpstan-return Collection<array-key, T>
+     */
+    private function createCollection($value): Collection
```

Comment:
I know I talked about a private function, but
this function is now implemented twice, should we extract this in a service instead ?

Reply:
I agree that we should not repeat the same logic, but I do not know if a service worths enough for this very specific purpose.

Reply:
I've created a trait in order to avoid duplicated code, but now I'm facing an [issue](https://github.com/sonata-project/SonataAdminBundle/pull/6315/checks?check_run_id=1207307156#step:4:188) with the generic annotations. Do you have any clue?